source/codegen/metadata/nixnet/custom_proto.mako

```diff
@@ -238,29 +238,66 @@ message DbSetPropertyResponse {
   int32 status = 1;
 }
 
-message Frame {
+message FrameRequest {
   uint64 timestamp = 1;
   uint32 identifier = 2;
-  uint32 type = 3;
-  uint32 flags = 4;
-  uint32 info = 5;
-  bytes payload = 6;
+  oneof type_enum{
+     FrameType type = 3;
+     uint32 type_raw = 4;
+  }
+  repeated FrameFlags flags = 5;
```

Comment:
I actually think it would be better to have `flags_raw` as a separate field not in a `oneof`. Since this is a logical OR we can simply OR together all flags from the FrameFlags array and the value of flags_raw. Of course most users will set one or the other and will work fine.

I think `bitfield`s are a very good candidate for the `_raw` concept. Some users will want to use existing code that `|` flags together and it's more difficult to "fake `_raw` as you can often do with typecasts on enums. Also, bitfields are often designed for extensibility, so there is a likelihood of "values we don't know about" that users might want to set through raw.

Reply:
I think I'm also going to add codegen for bitfield inputs for nxsocket as well. fyi.

Reply:
I think it's fine to add this later if implementation cost is a problem. You can check with Harsha. I believe that it would be easy to add later, especially with this being all custom_code. I think having a second non-oneof `_raw` field should be the "standard" implementation and that's what I'll do for codegen. But it's OK with me to have a functional subset of the "standard" implementation.

Reply:
Will be adding this implementation in later PRs